source/model/Activator.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
+
+
+class Activator(ABC):
+
+    def __init__(self, id):
+        self.activatorId = id
+
+    @property
+    @abstractmethod
+    def name(self):
+        """ The name of the activator which will be shown on the button"""
+        pass
+
+    @property
+    @abstractmethod
+    def stateType(self):
+        """ The type a state has. For example int, bool"""
+        pass
+
+    @property
+    @abstractmethod
+    def requiredInfo(self):
+        """
+        The information needed to operate the activator
+
+        This should be a dict that with a key value pairs
+        """
+        pass
+
+    @requiredInfo.setter
+    @abstractmethod
+    def requiredInfo(self, value):
```

Comment:
Do you prefer this as this is not really the pythonic way.

Could you please elaborate  on why this is cleaner in issue #18.